### PR TITLE
Synchronize PyTorchLightning/pytorch-lightning (revision 9aaa6822@master) to github/third-party/PyTorchLightning/pytorch-lightning

### PR DESCRIPTION
**Pull request by JiesiZhao077**
Summary:
### Manual
Patch P425611360, similar to D29307539:

- Patched D29251683. This is needed until https://github.com/PyTorchLightning/pytorch-lightning/pull/8044 or equivalent is merged upstream
- (ephemeral*) make ResultCollection._extract_batch_size a class method
- (ephemeral) patch pytorch_lightning/trainer/connectors/data_connector.py (https://github.com/PyTorchLightning/pytorch-lightning/pull/8087)
- ephemeral means that the change are made in-place in Lightning and will disappear after another sync.
- Update requirements.py

### New commit log messages
  9aaa6822 Add CODEOWNERS for progress dataclasses (#8196)
  bae08514 [refactor] Add should_raise_exception for gpus / tpus utilities (#8194)
  df6885cd add how to contribute (#8129)
  571a810a Improvements and changes to progress tracking dataclasses (#8140)
  2a7fad92 Avoid passing unnecessary params from TPUSpawn to DDPSpawn (#8192)
  f60aae98 Update `dataloaders` params in example (#8191)
  6db0fe36 training loop refactor - move val loop  (#8120)
  ae34df00 remove deadcode in trainer (#8121)
  b12a0d0a Make Plugins Proxies after transfering ownership (#8117)
  d6435a5b Bugfix/swa iterable dset (#8172)
  b1d8840f Fix metric attribute lookup (#8181)
  bf54ac1c fix NCCL error with non-consecutive trainer gpus (#8165)
  2f3c65e5 XLA Profiler integration (#8014)
  c521624a [bugfix] Add mechanism to prevent deadlock for DDP on Exception Trigger (#8167)
  1f025789 [bugfix] Clean Validation Sanity Checking metrics (#8171)
  c4492ad6 Merge pull request #8174 from PyTorchLightning/bugfix/8159_log_gpu_memory_on_step
  2a372e36 Fix module dict in base finetuning (#8170)
  b978d2a1 remove message (#8163)
  03154eb3 Refactor unnecessary `else` / `elif` when `if` block has a `return` statement (#8156)
  67f7e131 Fix dangerous default argument (#8164)
  9bd3747c Remove unnecessary use of comprehension (#8147)
  c3065c5c Iterate dictionary directly (#8155)
  51ea8422 resurface lost ddp info message (#8111)
  28afc7a1 ignore tests in DeepSource analyses (#8151)
  e11fe196 Remove unnecessary use of comprehension (#8149)
  3526d932 add .deepsource.toml (#8144)
  24db9140 Support state restoration of logged results 2/2(#7966)
  ad957108 document exceptions in utilities (#8122)
  55a90af7 `pytorch_lightning.loops` file structure: group by dataloader, epoch, and batch loop  (#8077)
  2c43bfc5 GPU CI - run torch 1.8 (LTS) (#8116)
  d4d5418c Fix notebook links (#8089)
  4d9b72b8 Nuke RPC (#8101)
  8bd7b1bd Add torchelastic check when sanitizing GPUs (#8095)
  4dc08e40 Loop Refactor 6/N - Remove Old Predict Loop (#8094)
  fe482031 restrict public interface of training loop (#8024)
  a45ab00b Loop Refactor 5/N - Prediction Loop (#7700)
  58b47dab Add `log_device_info` to Trainer (#8079)
  54ac4e03 Update fit with no validation hook test (#7738)
  06f83492 Support calling fit and test scripts using "python -m" module syntax with DDP (#8073)
  b378806b Add `add_to_queue`/`get_from_queue` for DDP spawn(#7916)
  6dd7797c Deprecate moved warning functions (#8085)
  8bce3943 Use XLA utility API to move data to CPU (Single TPU core) (#8078)

Reviewed By: kandluis

Differential Revision: D29459681

### CODE REVIEW

**Comment by ytsheng** [comment]
diff abandoned.
